packages/@lightningjs/ui-components/src/components/NavigationManager/NavigationManager.js

```diff
@@ -273,12 +273,28 @@ export default class NavigationManager extends FocusManager {
   _withAfterUpdate(component) {
     component.onAfterUpdate = function (element) {
       let hasChanged = false;
-      const watchProps = ['w', 'h', 'x', 'y', 'innerW', 'innerH'];
+      const watchProps = [
+        this._directionPropNames.crossAxis,
+        'w',
+        'h',
+        'innerW',
+        'innerH'
+      ];
 
       watchProps.forEach(prop => {
         const prevValueKey = `_navItemPrev${prop}`;
-        if (!hasChanged && element[prop] !== element[prevValueKey]) {
-          element[prevValueKey] = element[prop];
+
+        const propTransition = element.transition(prop);
+        const nextValue = propTransition
+          ? propTransition.targetValue
+          : element[prop];
+
+        if (propTransition.isRunning()) {
+          return;
+        }
+
+        if (nextValue !== element[prevValueKey]) {
+          element[prevValueKey] = nextValue;
```

Comment:
do you think any of this can/should be lifted out into a util we utilize later on?

Reply:
I think that could be useful. I can move this to utils.

Reply:
I'll do that on a separate branch. I think doing that abstraction here could make reviewing the changes in this diff more difficult to see.